source/frontend/src/components/maps/leaflet/Layers/PointClusterer.tsx

```diff
@@ -31,115 +28,87 @@ export type PointClustererProps = {
 
-/**
- * Converts the flat list of properties into the correct type of inventory property.
- * @param property A flat list of property values (from a Feature).
- */
-export const convertToProperty = (
-  property: any,
-  latitude?: number,
-  longitude?: number,
-): IProperty | null => {
-  return {
-    pid: property.PID,
-    latitude: latitude,
-    longitude: longitude,
-    address: {
-      id: property.ADDRESS_ID,
-      municipality: property.MUNICIPALITY_NAME,
-      provinceId: 1,
-      province: property.PROVINCE_STATE_CODE,
-      streetAddress1: property.STREET_ADDRESS_1,
-      postal: property.POSTAL_CODE,
-      country: property.COUNTRY_CODE,
-    },
-    pin: property.PIN,
-    landArea: property.LAND_AREA,
-    landLegalDescription: property.LAND_LEGAL_DESCRIPTION,
-    name: property.NAME,
-    description: property.DESCRIPTION,
-    isSensitive: property.IS_SENSITIVE,
-    isOwned: property.IS_OWNED,
-    encumbranceReason: property.ENCUMBRANCE_REASON,
-    isPropertyOfInterest: property.IS_PROPERTY_OF_INTEREST,
-    isPayableLease: property.IS_PAYABLE_LEASE,
-    isVisibleToOtherAgencies: property.IS_VISIBLE_TO_OTHER_AGENCIES,
-    areaUnit: property.PROPERTY_AREA_UNIT_TYPE_CODE,
-    classificationId: property.PROPERTY_CLASSIFICATION_TYPE_CODE,
-    id: property.PROPERTY_ID,
-    status: property.PROPERTY_STATUS_TYPE_CODE,
-    tenure: property.PROPERTY_TENURE_TYPE_CODE,
-    regionId: property.REGION_CODE,
-    zoning: property.ZONING,
-    zoningPotential: property.ZONING_POTENTIAL,
-  };
+  tilesLoaded: boolean;
 };
 
 /**
  * Clusters pins that are close together geographically based on the zoom level into a single clustered object.
  * @param param0 Point cluster properties.
  */
 export const PointClusterer: React.FC<React.PropsWithChildren<PointClustererProps>> = ({
-  points,
-  draftPoints,
   bounds,
   zoom,
-  onMarkerClick,
-  minZoom,
-  maxZoom,
+  minZoom: minZoomProps,
+  maxZoom: maxZoomProps,
   zoomToBoundsOnClick = true,
   spiderfyOnMaxZoom = true,
   tilesLoaded,
 }) => {
   // state and refs
-  const spiderfierRef = useRef<Spiderfier>();
+  const spiderfierRef =
+    useRef<Spiderfier<PIMS_Property_Location_View | PMBC_FullyAttributed_Feature_Properties>>();
   const featureGroupRef = useRef<L.FeatureGroup>(null);
   const draftFeatureGroupRef = useRef<L.FeatureGroup>(null);
   const filterState = useFilterContext();
-  const mapStateContext = React.useContext(MapStateContext);
-  const { selectedInventoryProperty: selected } = mapStateContext;
 
-  const [currentSelected, setCurrentSelected] = useState(selected);
-  const [currentCluster, setCurrentCluster] = useState<
-    ICluster<any, Supercluster.AnyProps> | undefined
-  >(undefined);
+  // TODO: Figure out if the currentCluster is needed
+  const [, setCurrentCluster] = useState<ICluster<any, Supercluster.AnyProps> | undefined>(
+    undefined,
+  );
+
+  const mapMachine = useMapStateMachine();
+
+  const selectedFeature = useMemo(() => {
```

Comment:
why is a memo needed here? this is just assigning a variable.

Reply:
fair point